tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -376,351 +384,218 @@ stages:
-            # - buildVersionToDownload: specific
-            # - buildId: <the id of the triggering build>
-            # preferTriggeringPipeline: true
-            source: specific
-            project: internal
-            pipeline: ${{ variables.artifactPipeline }}
-            buildVersionToDownload: specific
-            buildId: ${{ variables.artifactBuildId }}
-            artifact: pack
-            patterns: "**/${{ variables.escapedTestPackage }}-?.?.?-*.tgz"
-            path: $(Pipeline.Workspace)/client/pack
-            # allowPartiallySucceededBuilds: true # No effect as long as we have buildVersionToDownload: specific
-            # branchName: $(Build.SourceBranch)   # No effect as long as we have buildVersionToDownload: specific
-
-        # Install e2e test package
-        - task: Bash@3
-          displayName: Install Test Package
-          inputs:
-            targetType: 'inline'
-            workingDirectory: ${{ variables.testWorkspace }}
-            script: |
-              echo "Install ${{ variables.testPackage }}"
-              TEST_PACKAGE_FILE_PATTERN=${{ variables.escapedTestPackage }}-?.?.?-*.tgz
-              TEST_PACKAGE_PATH_PATTERN=$(Pipeline.Workspace)/client/pack/scoped/$TEST_PACKAGE_FILE_PATTERN
-
-              echo $TEST_PACKAGE_FILE_PATTERN
-              echo $TEST_PACKAGE_PATH_PATTERN
-
-              echo `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l`
-              echo `ls $TEST_PACKAGE_PATH_PATTERN`
-
-              if [[ `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l` -eq 1 ]]; then
-                TEST_PACKAGE_TGZ=`ls $TEST_PACKAGE_PATH_PATTERN`
-              else
-                ls -1 $TEST_PACKAGE_PATH_PATTERN
-                echo "##vso[task.logissue type=error]Test package '${{ variables.testPackage }}' not found, or there are more then one found"
-                exit -1
-              fi
-
-              npm install $TEST_PACKAGE_TGZ
-
-        - task: Bash@3
-          displayName: Prepare test package to run tests
-          inputs:
-            targetType: 'inline'
-            workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ variables.testPackage }}
-            script: |
-              cp ${{ variables.testWorkspace }}/.npmrc . ;
-              npm i ;
-
-        # We run both types of tests in the same bash step so we can make sure to run the second set even if the first
-        # one fails.
-        # Doing this with separate ADO steps is not easy.
-        # This step reports failure if either of the test runs reports failure.
-        - task: Bash@3
-          displayName: Run tests
-          inputs:
-            targetType: 'inline'
-            workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ variables.testPackage }}
-            script: |
-
-              # Run execution time tests
-              npm run test:benchmark:report;
-              executionTimeTestsExitCode=$?;
+  - ${{ each stage in parameters.endpoints }}:
```

Comment:
nit: consider a different variable name since "stage" is a keyword in pipelines. Maybe `stageName`?

Reply:
I changed to endpoint